vignettes/articles/oauth.Rmd

```diff
@@ -14,24 +14,61 @@ library(httr2)
+The API gives this permission in the form of an **access token**, which is essentially a random string of numbers and letters.
+An access token tends to be very short lived, lasting maybe only days or hours, and is bound to a specified **scope** of access.
+The access token has some big advantages over a user name and password:
+
+-   It's short-lived so if it's lost or stolen, there's only a limited amount of time where it can be abused.
+-   It has limited scope, so even if stolen, it can't be used to do something particularly nefarious like changing your password or contact details.
+-   It's bound to a specific application, so it can be invalidated (cancelled) without affecting any other uses.
+
+If you have an access token you can use it to authenticate an API by passing a **bearer token** in the `Authorization` header, which you can do with `req_auth_bearer_token()`.
+However, in most cases you will want to let httr2 manage this by calling one of the `req_oauth_` functions we'll talk about shortly.
+
+One of the reasons that you want httr2 to manage your tokens is that because access tokens are so short lived, they're often accompanied by a **refresh token**.
+A refresh token lasts for a longer amount of time and has one job: it allows you to get a new access token when the previous one expires.
+This leads to a hierarchy of credentials from strongest to weakest:
+
+-   user name + password: your code should never touch these!
+-   refresh token: it's okay to store this on your computer.
+-   access token: you submit this with every request.
+
+Now that you've got the basic idea of OAuth, lets talk about the details.
 
 ## Clients
 
-The first step in working with any OAuth API is to create an **application** **client**, which I'll generally just call a **client**.
-You'll do this by registering for a developer account on the API's website then creating a new OAuth app.
-This process varies from API to API (it's normal to spend some time hunting through the docs and settings), but at the end of it you'll get a client id (another random string of numbers and letters).
+The first step in working with any OAuth API is to create an **application** **client**.
+It's called an application client because in the wider world OAuth is usually used with a web, phone, or tv app, but here your "app" is going to be your R package.
+For this reason, httr2 calls the application client a **client**, but many APIs will call it an OAuth application.
 
-Sometimes the client id is all you need, and you can give httr2 the information about the client with a call to `oauth_client()`, something like this:
+To create a client, you'll need to first register for a developer account on the API's website.
+In most cases this is easy, totally automated, and only takes a couple of minutes.
+That will give you access to some sort of developer portal which you can then use to create a new OAuth app (aka a client).
```

Comment:
to create -> to register (create sounds too much like actually building the thing)